{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Splicing the mind: Master's Thesis\n",
    "Preprocessing pipeline for DEXSeq analysis between schizophrenia patients and contols in the nucleus accumbens"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "This notebook is for querying the OmicSoft DiseaseLAnd data source and preparing data to be in an adequate format for DEXSeq analysis in R."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import necessary libraries\n",
    "from opi import *\n",
    "\n",
    "datastore = OPI()\n",
    "\n",
    "import pandas as pd\n",
    "import math\n",
    "\n",
    "# Always display only 5 rows in this NB\n",
    "pd.set_option('display.max_rows', 5)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Querying the data"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Metadata extraction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_index</th>\n",
       "      <th>age_summary</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>tissue</th>\n",
       "      <th>subject_id</th>\n",
       "      <th>disease</th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85772</td>\n",
       "      <td>31 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X2353</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85966</td>\n",
       "      <td>31 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4463</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>85819</td>\n",
       "      <td>58 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4114</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>86046</td>\n",
       "      <td>50 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4404</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>221007</td>\n",
       "      <td>44 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_933</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>85907</td>\n",
       "      <td>36 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X3410</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>85942</td>\n",
       "      <td>45 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X2950</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>85908</td>\n",
       "      <td>24 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4385</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>220944</td>\n",
       "      <td>49 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_1088</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>86033</td>\n",
       "      <td>50 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X2384</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>85987</td>\n",
       "      <td>42 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X3525</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>221003</td>\n",
       "      <td>41 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_843</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>85988</td>\n",
       "      <td>50 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X3993</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>220952</td>\n",
       "      <td>58 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_1188</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>220981</td>\n",
       "      <td>42 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_1455</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>85823</td>\n",
       "      <td>34 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4619</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>220971</td>\n",
       "      <td>59 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_13175</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>220984</td>\n",
       "      <td>32 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_1581</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>85814</td>\n",
       "      <td>63 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X3684</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>85886</td>\n",
       "      <td>43 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4656</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>86047</td>\n",
       "      <td>55 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4413</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>221004</td>\n",
       "      <td>33 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_878</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>85775</td>\n",
       "      <td>53 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X2976</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>86045</td>\n",
       "      <td>52 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4357</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>85828</td>\n",
       "      <td>44 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X2956</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>221005</td>\n",
       "      <td>47 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_930</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>86044</td>\n",
       "      <td>36 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4296</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>220963</td>\n",
       "      <td>62 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_13017</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>85820</td>\n",
       "      <td>26 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X4301</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>220961</td>\n",
       "      <td>48 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_1296</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>85909</td>\n",
       "      <td>40 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE80655_X3651</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>220968</td>\n",
       "      <td>56 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>GSE202537_13071</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_index age_summary  disease_state ethnicity  gender  \\\n",
       "0          85772    31 years  schizophrenia     White  female   \n",
       "1          85966    31 years  schizophrenia     White    male   \n",
       "2          85819    58 years  schizophrenia     White  female   \n",
       "3          86046    50 years  schizophrenia     White    male   \n",
       "4         221007    44 years  schizophrenia     White    male   \n",
       "5          85907    36 years  schizophrenia     White    male   \n",
       "6          85942    45 years  schizophrenia     White    male   \n",
       "7          85908    24 years  schizophrenia     White    male   \n",
       "8         220944    49 years  schizophrenia     White    male   \n",
       "9          86033    50 years  schizophrenia     White    male   \n",
       "10         85987    42 years  schizophrenia     White    male   \n",
       "11        221003    41 years  schizophrenia     White  female   \n",
       "12         85988    50 years  schizophrenia     White    male   \n",
       "13        220952    58 years  schizophrenia     White    male   \n",
       "14        220981    42 years  schizophrenia     White    male   \n",
       "15         85823    34 years  schizophrenia     White    male   \n",
       "16        220971    59 years  schizophrenia     White    male   \n",
       "17        220984    32 years  schizophrenia     White    male   \n",
       "18         85814    63 years  schizophrenia     White    male   \n",
       "19         85886    43 years  schizophrenia     White    male   \n",
       "20         86047    55 years  schizophrenia     White    male   \n",
       "21        221004    33 years  schizophrenia     White    male   \n",
       "22         85775    53 years  schizophrenia     White  female   \n",
       "23         86045    52 years  schizophrenia     White    male   \n",
       "24         85828    44 years  schizophrenia     White    male   \n",
       "25        221005    47 years  schizophrenia     White    male   \n",
       "26         86044    36 years  schizophrenia     White    male   \n",
       "27        220963    62 years  schizophrenia     White    male   \n",
       "28         85820    26 years  schizophrenia     White    male   \n",
       "29        220961    48 years  schizophrenia     White    male   \n",
       "30         85909    40 years  schizophrenia     White    male   \n",
       "31        220968    56 years  schizophrenia     White    male   \n",
       "\n",
       "               tissue       subject_id             disease  \\\n",
       "0   nucleus accumbens   GSE80655_X2353     mental disorder   \n",
       "1   nucleus accumbens   GSE80655_X4463     mental disorder   \n",
       "2   nucleus accumbens   GSE80655_X4114     mental disorder   \n",
       "3   nucleus accumbens   GSE80655_X4404     mental disorder   \n",
       "4   nucleus accumbens    GSE202537_933  psychotic disorder   \n",
       "5   nucleus accumbens   GSE80655_X3410     mental disorder   \n",
       "6   nucleus accumbens   GSE80655_X2950     mental disorder   \n",
       "7   nucleus accumbens   GSE80655_X4385     mental disorder   \n",
       "8   nucleus accumbens   GSE202537_1088  psychotic disorder   \n",
       "9   nucleus accumbens   GSE80655_X2384     mental disorder   \n",
       "10  nucleus accumbens   GSE80655_X3525     mental disorder   \n",
       "11  nucleus accumbens    GSE202537_843  psychotic disorder   \n",
       "12  nucleus accumbens   GSE80655_X3993     mental disorder   \n",
       "13  nucleus accumbens   GSE202537_1188  psychotic disorder   \n",
       "14  nucleus accumbens   GSE202537_1455  psychotic disorder   \n",
       "15  nucleus accumbens   GSE80655_X4619     mental disorder   \n",
       "16  nucleus accumbens  GSE202537_13175  psychotic disorder   \n",
       "17  nucleus accumbens   GSE202537_1581  psychotic disorder   \n",
       "18  nucleus accumbens   GSE80655_X3684     mental disorder   \n",
       "19  nucleus accumbens   GSE80655_X4656     mental disorder   \n",
       "20  nucleus accumbens   GSE80655_X4413     mental disorder   \n",
       "21  nucleus accumbens    GSE202537_878  psychotic disorder   \n",
       "22  nucleus accumbens   GSE80655_X2976     mental disorder   \n",
       "23  nucleus accumbens   GSE80655_X4357     mental disorder   \n",
       "24  nucleus accumbens   GSE80655_X2956     mental disorder   \n",
       "25  nucleus accumbens    GSE202537_930  psychotic disorder   \n",
       "26  nucleus accumbens   GSE80655_X4296     mental disorder   \n",
       "27  nucleus accumbens  GSE202537_13017  psychotic disorder   \n",
       "28  nucleus accumbens   GSE80655_X4301     mental disorder   \n",
       "29  nucleus accumbens   GSE202537_1296  psychotic disorder   \n",
       "30  nucleus accumbens   GSE80655_X3651     mental disorder   \n",
       "31  nucleus accumbens  GSE202537_13071  psychotic disorder   \n",
       "\n",
       "                                                title  \n",
       "0   RNA-sequencing of human post-mortem brain tissues  \n",
       "1   RNA-sequencing of human post-mortem brain tissues  \n",
       "2   RNA-sequencing of human post-mortem brain tissues  \n",
       "3   RNA-sequencing of human post-mortem brain tissues  \n",
       "4   Diurnal alterations in gene expression across ...  \n",
       "5   RNA-sequencing of human post-mortem brain tissues  \n",
       "6   RNA-sequencing of human post-mortem brain tissues  \n",
       "7   RNA-sequencing of human post-mortem brain tissues  \n",
       "8   Diurnal alterations in gene expression across ...  \n",
       "9   RNA-sequencing of human post-mortem brain tissues  \n",
       "10  RNA-sequencing of human post-mortem brain tissues  \n",
       "11  Diurnal alterations in gene expression across ...  \n",
       "12  RNA-sequencing of human post-mortem brain tissues  \n",
       "13  Diurnal alterations in gene expression across ...  \n",
       "14  Diurnal alterations in gene expression across ...  \n",
       "15  RNA-sequencing of human post-mortem brain tissues  \n",
       "16  Diurnal alterations in gene expression across ...  \n",
       "17  Diurnal alterations in gene expression across ...  \n",
       "18  RNA-sequencing of human post-mortem brain tissues  \n",
       "19  RNA-sequencing of human post-mortem brain tissues  \n",
       "20  RNA-sequencing of human post-mortem brain tissues  \n",
       "21  Diurnal alterations in gene expression across ...  \n",
       "22  RNA-sequencing of human post-mortem brain tissues  \n",
       "23  RNA-sequencing of human post-mortem brain tissues  \n",
       "24  RNA-sequencing of human post-mortem brain tissues  \n",
       "25  Diurnal alterations in gene expression across ...  \n",
       "26  RNA-sequencing of human post-mortem brain tissues  \n",
       "27  Diurnal alterations in gene expression across ...  \n",
       "28  RNA-sequencing of human post-mortem brain tissues  \n",
       "29  Diurnal alterations in gene expression across ...  \n",
       "30  RNA-sequencing of human post-mortem brain tissues  \n",
       "31  Diurnal alterations in gene expression across ...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Query metadata of schizophrenia studies from the nucleus accumbens tissue and white ethnicity\n",
    "sql = \"\"\" \n",
    "WITH valid_samples AS (\n",
    "  SELECT DISTINCT sample_index\n",
    "  FROM human_disease_b38_gc33.exon_data\n",
    ")\n",
    "SELECT \n",
    "  s.sample_index,\n",
    "  s.age_summary,\n",
    "  s.disease_state,\n",
    "  s.ethnicity,\n",
    "  s.gender,\n",
    "  s.tissue,\n",
    "  s.subject_id,\n",
    "  p.disease,\n",
    "  p.title\n",
    "FROM human_disease_b38_gc33.samples s\n",
    "JOIN human_disease_b38_gc33.projects p \n",
    "  ON s.project_id = p.project_id\n",
    "WHERE s.disease_state IN 'schizophrenia'           # eaarlier: ('schizophrenia', 'schizoaffective disorder')\n",
    "  AND s.tissue = 'nucleus accumbens'\n",
    "  AND s.ethnicity = 'White'\n",
    "  AND s.sample_index IN (SELECT sample_index FROM valid_samples)\n",
    "  LIMIT 5\n",
    "\"\"\"\n",
    "sz_nacc_metadata = datastore.query(sql)\n",
    "display(sz_nacc_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "tissue\n",
       "nucleus accumbens    32\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 140,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sanity check\n",
    "sz_nacc_metadata[\"tissue\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "disease\n",
       "mental disorder       20\n",
       "psychotic disorder    12\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sz_nacc_metadata[\"disease\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "title\n",
       "RNA-sequencing of human post-mortem brain tissues                                 20\n",
       "Diurnal alterations in gene expression across striatal subregions in psychosis    12\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 142,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sz_nacc_metadata[\"title\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "title\n",
       "Diurnal alterations in gene expression across striatal subregions in psychosis    36\n",
       "RNA-sequencing of human post-mortem brain tissues                                 22\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_index</th>\n",
       "      <th>age_summary</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>tissue</th>\n",
       "      <th>disease</th>\n",
       "      <th>title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85954</td>\n",
       "      <td>63 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85953</td>\n",
       "      <td>32 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>220954</td>\n",
       "      <td>36 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>220996</td>\n",
       "      <td>38 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>220958</td>\n",
       "      <td>58 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>220979</td>\n",
       "      <td>45 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>220940</td>\n",
       "      <td>23 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>220988</td>\n",
       "      <td>49 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>85836</td>\n",
       "      <td>66 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>220942</td>\n",
       "      <td>49 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>220962</td>\n",
       "      <td>48 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>220977</td>\n",
       "      <td>43 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>220976</td>\n",
       "      <td>37 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>220970</td>\n",
       "      <td>46 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>220998</td>\n",
       "      <td>42 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>86039</td>\n",
       "      <td>56 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>85803</td>\n",
       "      <td>35 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>85941</td>\n",
       "      <td>44 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>85805</td>\n",
       "      <td>41 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>85998</td>\n",
       "      <td>58 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>220986</td>\n",
       "      <td>21 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>85883</td>\n",
       "      <td>40 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>85804</td>\n",
       "      <td>39 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>220964</td>\n",
       "      <td>52 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>220966</td>\n",
       "      <td>48 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>220978</td>\n",
       "      <td>51 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>220985</td>\n",
       "      <td>50 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>221000</td>\n",
       "      <td>40 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>220973</td>\n",
       "      <td>58 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>220982</td>\n",
       "      <td>50 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>85774</td>\n",
       "      <td>45 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>86002</td>\n",
       "      <td>48 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>85825</td>\n",
       "      <td>32 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>85837</td>\n",
       "      <td>49 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>85956</td>\n",
       "      <td>45 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>220946</td>\n",
       "      <td>24 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>85943</td>\n",
       "      <td>67 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>220993</td>\n",
       "      <td>39 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>220972</td>\n",
       "      <td>58 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>220983</td>\n",
       "      <td>45 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>220949</td>\n",
       "      <td>51 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>85925</td>\n",
       "      <td>70 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>220967</td>\n",
       "      <td>53 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>85884</td>\n",
       "      <td>55 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>220989</td>\n",
       "      <td>53 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>85802</td>\n",
       "      <td>70 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>220937</td>\n",
       "      <td>49 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>220950</td>\n",
       "      <td>56 years</td>\n",
       "      <td>disease control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_index age_summary    disease_state ethnicity  gender  \\\n",
       "0          85954    63 years   normal control     White    male   \n",
       "1          85953    32 years   normal control     White    male   \n",
       "2         220954    36 years  disease control     White  female   \n",
       "3         220996    38 years  disease control     White    male   \n",
       "4         220958    58 years  disease control     White  female   \n",
       "5         220979    45 years  disease control     White    male   \n",
       "6         220940    23 years  disease control     White  female   \n",
       "7         220988    49 years  disease control     White    male   \n",
       "8          85836    66 years   normal control     White    male   \n",
       "9         220942    49 years  disease control     White    male   \n",
       "10        220962    48 years  disease control     White    male   \n",
       "11        220977    43 years  disease control     White    male   \n",
       "12        220976    37 years  disease control     White    male   \n",
       "13        220970    46 years  disease control     White  female   \n",
       "14        220998    42 years  disease control     White    male   \n",
       "15         86039    56 years   normal control     White    male   \n",
       "16         85803    35 years   normal control     White    male   \n",
       "17         85941    44 years   normal control     White    male   \n",
       "18         85805    41 years   normal control     White    male   \n",
       "19         85998    58 years   normal control     White    male   \n",
       "20        220986    21 years  disease control     White  female   \n",
       "21         85883    40 years   normal control     White    male   \n",
       "22         85804    39 years   normal control     White    male   \n",
       "23        220964    52 years  disease control     White    male   \n",
       "24        220966    48 years  disease control     White    male   \n",
       "25        220978    51 years  disease control     White  female   \n",
       "26        220985    50 years  disease control     White    male   \n",
       "27        221000    40 years  disease control     White    male   \n",
       "28        220973    58 years  disease control     White    male   \n",
       "29        220982    50 years  disease control     White    male   \n",
       "30         85774    45 years   normal control     White    male   \n",
       "31         86002    48 years   normal control     White    male   \n",
       "32         85825    32 years   normal control     White    male   \n",
       "33         85837    49 years   normal control     White    male   \n",
       "34         85956    45 years   normal control     White  female   \n",
       "35        220946    24 years  disease control     White  female   \n",
       "36         85943    67 years   normal control     White    male   \n",
       "37        220993    39 years  disease control     White    male   \n",
       "38        220972    58 years  disease control     White    male   \n",
       "39        220983    45 years  disease control     White  female   \n",
       "40        220949    51 years  disease control     White    male   \n",
       "41         85925    70 years   normal control     White  female   \n",
       "42        220967    53 years  disease control     White    male   \n",
       "43         85884    55 years   normal control     White    male   \n",
       "44        220989    53 years  disease control     White  female   \n",
       "45         85802    70 years   normal control     White    male   \n",
       "46        220937    49 years  disease control     White    male   \n",
       "47        220950    56 years  disease control     White    male   \n",
       "\n",
       "               tissue             disease  \\\n",
       "0   nucleus accumbens     mental disorder   \n",
       "1   nucleus accumbens     mental disorder   \n",
       "2   nucleus accumbens  psychotic disorder   \n",
       "3   nucleus accumbens  psychotic disorder   \n",
       "4   nucleus accumbens  psychotic disorder   \n",
       "5   nucleus accumbens  psychotic disorder   \n",
       "6   nucleus accumbens  psychotic disorder   \n",
       "7   nucleus accumbens  psychotic disorder   \n",
       "8   nucleus accumbens     mental disorder   \n",
       "9   nucleus accumbens  psychotic disorder   \n",
       "10  nucleus accumbens  psychotic disorder   \n",
       "11  nucleus accumbens  psychotic disorder   \n",
       "12  nucleus accumbens  psychotic disorder   \n",
       "13  nucleus accumbens  psychotic disorder   \n",
       "14  nucleus accumbens  psychotic disorder   \n",
       "15  nucleus accumbens     mental disorder   \n",
       "16  nucleus accumbens     mental disorder   \n",
       "17  nucleus accumbens     mental disorder   \n",
       "18  nucleus accumbens     mental disorder   \n",
       "19  nucleus accumbens     mental disorder   \n",
       "20  nucleus accumbens  psychotic disorder   \n",
       "21  nucleus accumbens     mental disorder   \n",
       "22  nucleus accumbens     mental disorder   \n",
       "23  nucleus accumbens  psychotic disorder   \n",
       "24  nucleus accumbens  psychotic disorder   \n",
       "25  nucleus accumbens  psychotic disorder   \n",
       "26  nucleus accumbens  psychotic disorder   \n",
       "27  nucleus accumbens  psychotic disorder   \n",
       "28  nucleus accumbens  psychotic disorder   \n",
       "29  nucleus accumbens  psychotic disorder   \n",
       "30  nucleus accumbens     mental disorder   \n",
       "31  nucleus accumbens     mental disorder   \n",
       "32  nucleus accumbens     mental disorder   \n",
       "33  nucleus accumbens     mental disorder   \n",
       "34  nucleus accumbens     mental disorder   \n",
       "35  nucleus accumbens  psychotic disorder   \n",
       "36  nucleus accumbens     mental disorder   \n",
       "37  nucleus accumbens  psychotic disorder   \n",
       "38  nucleus accumbens  psychotic disorder   \n",
       "39  nucleus accumbens  psychotic disorder   \n",
       "40  nucleus accumbens  psychotic disorder   \n",
       "41  nucleus accumbens     mental disorder   \n",
       "42  nucleus accumbens  psychotic disorder   \n",
       "43  nucleus accumbens     mental disorder   \n",
       "44  nucleus accumbens  psychotic disorder   \n",
       "45  nucleus accumbens     mental disorder   \n",
       "46  nucleus accumbens  psychotic disorder   \n",
       "47  nucleus accumbens  psychotic disorder   \n",
       "\n",
       "                                                title  \n",
       "0   RNA-sequencing of human post-mortem brain tissues  \n",
       "1   RNA-sequencing of human post-mortem brain tissues  \n",
       "2   Diurnal alterations in gene expression across ...  \n",
       "3   Diurnal alterations in gene expression across ...  \n",
       "4   Diurnal alterations in gene expression across ...  \n",
       "5   Diurnal alterations in gene expression across ...  \n",
       "6   Diurnal alterations in gene expression across ...  \n",
       "7   Diurnal alterations in gene expression across ...  \n",
       "8   RNA-sequencing of human post-mortem brain tissues  \n",
       "9   Diurnal alterations in gene expression across ...  \n",
       "10  Diurnal alterations in gene expression across ...  \n",
       "11  Diurnal alterations in gene expression across ...  \n",
       "12  Diurnal alterations in gene expression across ...  \n",
       "13  Diurnal alterations in gene expression across ...  \n",
       "14  Diurnal alterations in gene expression across ...  \n",
       "15  RNA-sequencing of human post-mortem brain tissues  \n",
       "16  RNA-sequencing of human post-mortem brain tissues  \n",
       "17  RNA-sequencing of human post-mortem brain tissues  \n",
       "18  RNA-sequencing of human post-mortem brain tissues  \n",
       "19  RNA-sequencing of human post-mortem brain tissues  \n",
       "20  Diurnal alterations in gene expression across ...  \n",
       "21  RNA-sequencing of human post-mortem brain tissues  \n",
       "22  RNA-sequencing of human post-mortem brain tissues  \n",
       "23  Diurnal alterations in gene expression across ...  \n",
       "24  Diurnal alterations in gene expression across ...  \n",
       "25  Diurnal alterations in gene expression across ...  \n",
       "26  Diurnal alterations in gene expression across ...  \n",
       "27  Diurnal alterations in gene expression across ...  \n",
       "28  Diurnal alterations in gene expression across ...  \n",
       "29  Diurnal alterations in gene expression across ...  \n",
       "30  RNA-sequencing of human post-mortem brain tissues  \n",
       "31  RNA-sequencing of human post-mortem brain tissues  \n",
       "32  RNA-sequencing of human post-mortem brain tissues  \n",
       "33  RNA-sequencing of human post-mortem brain tissues  \n",
       "34  RNA-sequencing of human post-mortem brain tissues  \n",
       "35  Diurnal alterations in gene expression across ...  \n",
       "36  RNA-sequencing of human post-mortem brain tissues  \n",
       "37  Diurnal alterations in gene expression across ...  \n",
       "38  Diurnal alterations in gene expression across ...  \n",
       "39  Diurnal alterations in gene expression across ...  \n",
       "40  Diurnal alterations in gene expression across ...  \n",
       "41  RNA-sequencing of human post-mortem brain tissues  \n",
       "42  Diurnal alterations in gene expression across ...  \n",
       "43  RNA-sequencing of human post-mortem brain tissues  \n",
       "44  Diurnal alterations in gene expression across ...  \n",
       "45  RNA-sequencing of human post-mortem brain tissues  \n",
       "46  Diurnal alterations in gene expression across ...  \n",
       "47  Diurnal alterations in gene expression across ...  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Query normal control metadata - NAcc, white\n",
    "sql = \"\"\" \n",
    "WITH valid_samples AS (\n",
    "  SELECT DISTINCT s.sample_index\n",
    "  FROM human_disease_b38_gc33.samples s\n",
    "  JOIN human_disease_b38_gc33.projects p \n",
    "    ON s.project_id = p.project_id\n",
    "  JOIN human_disease_b38_gc33.exon_data ed\n",
    "    ON s.sample_index = ed.sample_index\n",
    "  WHERE s.disease_state IN ('normal control', 'disease control')\n",
    "    AND s.tissue = 'nucleus accumbens'\n",
    "    AND s.ethnicity = 'White'\n",
    "    AND p.title IN (\n",
    "         'RNA-sequencing of human post-mortem brain tissues', \n",
    "         'Diurnal alterations in gene expression across striatal subregions in psychosis'\n",
    "    )\n",
    ")\n",
    "SELECT \n",
    "  s.sample_index,\n",
    "  s.age_summary,\n",
    "  s.disease_state,\n",
    "  s.ethnicity,\n",
    "  s.gender,\n",
    "  s.tissue,\n",
    "  p.disease,\n",
    "  p.title\n",
    "FROM human_disease_b38_gc33.samples s\n",
    "JOIN human_disease_b38_gc33.projects p \n",
    "  ON s.project_id = p.project_id\n",
    "WHERE s.sample_index IN (SELECT sample_index FROM valid_samples)\n",
    "LIMIT 5\n",
    "\"\"\"\n",
    "normal_nacc_metadata = datastore.query(sql)\n",
    "display(normal_nacc_metadata)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "normal_nacc_metadata[\"title\"].value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Cleaning up the resulting dfs and merging them"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 189,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Extract numeric digits from the 'age_summary' column and convert to int\n",
    "normal_nacc_metadata['age'] = normal_nacc_metadata['age_summary'].str.extract(r'(\\d+)').astype(int)\n",
    "sz_nacc_metadata[\"age\"] = sz_nacc_metadata['age_summary'].str.extract(r'(\\d+)').astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>normal control</th>\n",
       "      <th>White</th>\n",
       "      <td>48.0</td>\n",
       "      <td>47.229167</td>\n",
       "      <td>11.255712</td>\n",
       "      <td>21.0</td>\n",
       "      <td>40.0</td>\n",
       "      <td>48.0</td>\n",
       "      <td>53.5</td>\n",
       "      <td>70.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          count       mean        std   min   25%   50%   75%  \\\n",
       "disease_state  ethnicity                                                        \n",
       "normal control White       48.0  47.229167  11.255712  21.0  40.0  48.0  53.5   \n",
       "\n",
       "                           max  \n",
       "disease_state  ethnicity        \n",
       "normal control White      70.0  "
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Set disease control to normal control\n",
    "normal_nacc_metadata[\"disease_state\"] = normal_nacc_metadata[\"disease_state\"].replace({\n",
    "    \"disease control\": \"normal control\"\n",
    "    })\n",
    "# Compute descriptive stats\n",
    "normal_nacc_metadata.groupby([\"disease_state\", \"ethnicity\"])[\"age\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "      <th>mean</th>\n",
       "      <th>std</th>\n",
       "      <th>min</th>\n",
       "      <th>25%</th>\n",
       "      <th>50%</th>\n",
       "      <th>75%</th>\n",
       "      <th>max</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>schizophrenia</th>\n",
       "      <th>White</th>\n",
       "      <td>32.0</td>\n",
       "      <td>44.8125</td>\n",
       "      <td>10.5324</td>\n",
       "      <td>24.0</td>\n",
       "      <td>36.0</td>\n",
       "      <td>44.5</td>\n",
       "      <td>52.25</td>\n",
       "      <td>63.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                         count     mean      std   min   25%   50%    75%  \\\n",
       "disease_state ethnicity                                                     \n",
       "schizophrenia White       32.0  44.8125  10.5324  24.0  36.0  44.5  52.25   \n",
       "\n",
       "                          max  \n",
       "disease_state ethnicity        \n",
       "schizophrenia White      63.0  "
      ]
     },
     "execution_count": 191,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Compute descriptive stats\n",
    "sz_nacc_metadata.groupby([\"disease_state\", \"ethnicity\"])[\"age\"].describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count    48.000000\n",
      "mean     47.229167\n",
      "std      11.255712\n",
      "min      21.000000\n",
      "25%      40.000000\n",
      "50%      48.000000\n",
      "75%      53.500000\n",
      "max      70.000000\n",
      "Name: age, dtype: float64\n",
      "gender\n",
      "male      37\n",
      "female    11\n",
      "Name: count, dtype: Int64\n"
     ]
    }
   ],
   "source": [
    "print(normal_nacc_metadata[\"age\"].describe())\n",
    "print(normal_nacc_metadata[\"gender\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 193,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "count    32.0000\n",
      "mean     44.8125\n",
      "std      10.5324\n",
      "min      24.0000\n",
      "25%      36.0000\n",
      "50%      44.5000\n",
      "75%      52.2500\n",
      "max      63.0000\n",
      "Name: age, dtype: float64\n",
      "gender\n",
      "male      28\n",
      "female     4\n",
      "Name: count, dtype: Int64\n"
     ]
    }
   ],
   "source": [
    "print(sz_nacc_metadata[\"age\"].describe())\n",
    "print(sz_nacc_metadata[\"gender\"].value_counts())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_index</th>\n",
       "      <th>age_summary</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>tissue</th>\n",
       "      <th>disease</th>\n",
       "      <th>title</th>\n",
       "      <th>age</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85954</td>\n",
       "      <td>63 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>63</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85953</td>\n",
       "      <td>32 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>32</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>220954</td>\n",
       "      <td>36 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>36</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>220996</td>\n",
       "      <td>38 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>38</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>220958</td>\n",
       "      <td>58 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>58</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>220963</td>\n",
       "      <td>62 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>62</td>\n",
       "      <td>GSE202537_13017</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>85820</td>\n",
       "      <td>26 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>26</td>\n",
       "      <td>GSE80655_X4301</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>220961</td>\n",
       "      <td>48 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>48</td>\n",
       "      <td>GSE202537_1296</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>85909</td>\n",
       "      <td>40 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>40</td>\n",
       "      <td>GSE80655_X3651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>220968</td>\n",
       "      <td>56 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>56</td>\n",
       "      <td>GSE202537_13071</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_index age_summary   disease_state ethnicity  gender  \\\n",
       "0          85954    63 years  normal control     White    male   \n",
       "1          85953    32 years  normal control     White    male   \n",
       "2         220954    36 years  normal control     White  female   \n",
       "3         220996    38 years  normal control     White    male   \n",
       "4         220958    58 years  normal control     White  female   \n",
       "..           ...         ...             ...       ...     ...   \n",
       "27        220963    62 years   schizophrenia     White    male   \n",
       "28         85820    26 years   schizophrenia     White    male   \n",
       "29        220961    48 years   schizophrenia     White    male   \n",
       "30         85909    40 years   schizophrenia     White    male   \n",
       "31        220968    56 years   schizophrenia     White    male   \n",
       "\n",
       "               tissue             disease  \\\n",
       "0   nucleus accumbens     mental disorder   \n",
       "1   nucleus accumbens     mental disorder   \n",
       "2   nucleus accumbens  psychotic disorder   \n",
       "3   nucleus accumbens  psychotic disorder   \n",
       "4   nucleus accumbens  psychotic disorder   \n",
       "..                ...                 ...   \n",
       "27  nucleus accumbens  psychotic disorder   \n",
       "28  nucleus accumbens     mental disorder   \n",
       "29  nucleus accumbens  psychotic disorder   \n",
       "30  nucleus accumbens     mental disorder   \n",
       "31  nucleus accumbens  psychotic disorder   \n",
       "\n",
       "                                                title  age       subject_id  \n",
       "0   RNA-sequencing of human post-mortem brain tissues   63             <NA>  \n",
       "1   RNA-sequencing of human post-mortem brain tissues   32             <NA>  \n",
       "2   Diurnal alterations in gene expression across ...   36             <NA>  \n",
       "3   Diurnal alterations in gene expression across ...   38             <NA>  \n",
       "4   Diurnal alterations in gene expression across ...   58             <NA>  \n",
       "..                                                ...  ...              ...  \n",
       "27  Diurnal alterations in gene expression across ...   62  GSE202537_13017  \n",
       "28  RNA-sequencing of human post-mortem brain tissues   26   GSE80655_X4301  \n",
       "29  Diurnal alterations in gene expression across ...   48   GSE202537_1296  \n",
       "30  RNA-sequencing of human post-mortem brain tissues   40   GSE80655_X3651  \n",
       "31  Diurnal alterations in gene expression across ...   56  GSE202537_13071  \n",
       "\n",
       "[80 rows x 10 columns]"
      ]
     },
     "execution_count": 194,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Merge metadatas\n",
    "metadata_nacc = pd.concat([normal_nacc_metadata, sz_nacc_metadata], axis=0)\n",
    "metadata_nacc.head()\n",
    "shape(metadata_nacc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_index</th>\n",
       "      <th>age_summary</th>\n",
       "      <th>disease_state</th>\n",
       "      <th>ethnicity</th>\n",
       "      <th>gender</th>\n",
       "      <th>tissue</th>\n",
       "      <th>disease</th>\n",
       "      <th>title</th>\n",
       "      <th>age</th>\n",
       "      <th>subject_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85772</td>\n",
       "      <td>31 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>31</td>\n",
       "      <td>GSE80655_X2353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>85774</td>\n",
       "      <td>45 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>45</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>85775</td>\n",
       "      <td>53 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>53</td>\n",
       "      <td>GSE80655_X2976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>85802</td>\n",
       "      <td>70 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>70</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>85803</td>\n",
       "      <td>35 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>mental disorder</td>\n",
       "      <td>RNA-sequencing of human post-mortem brain tissues</td>\n",
       "      <td>35</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>221000</td>\n",
       "      <td>40 years</td>\n",
       "      <td>normal control</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>40</td>\n",
       "      <td>&lt;NA&gt;</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>221003</td>\n",
       "      <td>41 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>female</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>41</td>\n",
       "      <td>GSE202537_843</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>221004</td>\n",
       "      <td>33 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>33</td>\n",
       "      <td>GSE202537_878</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>221005</td>\n",
       "      <td>47 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>47</td>\n",
       "      <td>GSE202537_930</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>221007</td>\n",
       "      <td>44 years</td>\n",
       "      <td>schizophrenia</td>\n",
       "      <td>White</td>\n",
       "      <td>male</td>\n",
       "      <td>nucleus accumbens</td>\n",
       "      <td>psychotic disorder</td>\n",
       "      <td>Diurnal alterations in gene expression across ...</td>\n",
       "      <td>44</td>\n",
       "      <td>GSE202537_933</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>80 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    sample_index age_summary   disease_state ethnicity  gender  \\\n",
       "0          85772    31 years   schizophrenia     White  female   \n",
       "30         85774    45 years  normal control     White    male   \n",
       "22         85775    53 years   schizophrenia     White  female   \n",
       "45         85802    70 years  normal control     White    male   \n",
       "16         85803    35 years  normal control     White    male   \n",
       "..           ...         ...             ...       ...     ...   \n",
       "27        221000    40 years  normal control     White    male   \n",
       "11        221003    41 years   schizophrenia     White  female   \n",
       "21        221004    33 years   schizophrenia     White    male   \n",
       "25        221005    47 years   schizophrenia     White    male   \n",
       "4         221007    44 years   schizophrenia     White    male   \n",
       "\n",
       "               tissue             disease  \\\n",
       "0   nucleus accumbens     mental disorder   \n",
       "30  nucleus accumbens     mental disorder   \n",
       "22  nucleus accumbens     mental disorder   \n",
       "45  nucleus accumbens     mental disorder   \n",
       "16  nucleus accumbens     mental disorder   \n",
       "..                ...                 ...   \n",
       "27  nucleus accumbens  psychotic disorder   \n",
       "11  nucleus accumbens  psychotic disorder   \n",
       "21  nucleus accumbens  psychotic disorder   \n",
       "25  nucleus accumbens  psychotic disorder   \n",
       "4   nucleus accumbens  psychotic disorder   \n",
       "\n",
       "                                                title  age      subject_id  \n",
       "0   RNA-sequencing of human post-mortem brain tissues   31  GSE80655_X2353  \n",
       "30  RNA-sequencing of human post-mortem brain tissues   45            <NA>  \n",
       "22  RNA-sequencing of human post-mortem brain tissues   53  GSE80655_X2976  \n",
       "45  RNA-sequencing of human post-mortem brain tissues   70            <NA>  \n",
       "16  RNA-sequencing of human post-mortem brain tissues   35            <NA>  \n",
       "..                                                ...  ...             ...  \n",
       "27  Diurnal alterations in gene expression across ...   40            <NA>  \n",
       "11  Diurnal alterations in gene expression across ...   41   GSE202537_843  \n",
       "21  Diurnal alterations in gene expression across ...   33   GSE202537_878  \n",
       "25  Diurnal alterations in gene expression across ...   47   GSE202537_930  \n",
       "4   Diurnal alterations in gene expression across ...   44   GSE202537_933  \n",
       "\n",
       "[80 rows x 10 columns]"
      ]
     },
     "execution_count": 195,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Sort sample_indexes\n",
    "metadata_nacc = metadata_nacc.sort_values(by=\"sample_index\")\n",
    "metadata_nacc.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save metadata to csv\n",
    "metadata_nacc.to_csv('metadata_nacc_.csv', index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Prepare count martix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Sample Indexes: [85772, 85774, 85775, 85802, 85803, 85804, 85805, 85814, 85819, 85820, 85823, 85825, 85828, 85836, 85837, 85883, 85884, 85886, 85907, 85908, 85909, 85925, 85941, 85942, 85943, 85953, 85954, 85956, 85966, 85987, 85988, 85998, 86002, 86033, 86039, 86044, 86045, 86046, 86047, 220937, 220940, 220942, 220944, 220946, 220949, 220950, 220952, 220954, 220958, 220961, 220962, 220963, 220964, 220966, 220967, 220968, 220970, 220971, 220972, 220973, 220976, 220977, 220978, 220979, 220981, 220982, 220983, 220984, 220985, 220986, 220988, 220989, 220993, 220996, 220998, 221000, 221003, 221004, 221005, 221007]\n",
      "80\n"
     ]
    }
   ],
   "source": [
    "# Filter exon_data for those samples present in metadata_nacc\n",
    "# Get the unique sample indexes from metadata_nacc to use for query \n",
    "sample_idx = metadata_nacc['sample_index'].unique().tolist()\n",
    "print((\"Sample Indexes:\", sample_idx)[0:5])\n",
    "print(len(sample_idx))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "85772,85774,85775,85802,85803,85804,85805,85814,85819,85820,85823,85825,85828,85836,85837,85883,85884,85886,85907,85908,85909,85925,85941,85942,85943,85953,85954,85956,85966,85987,85988,85998,86002,86033,86039,86044,86045,86046,86047,220937,220940,220942,220944,220946,220949,220950,220952,220954,220958,220961,220962,220963,220964,220966,220967,220968,220970,220971,220972,220973,220976,220977,220978,220979,220981,220982,220983,220984,220985,220986,220988,220989,220993,220996,220998,221000,221003,221004,221005,221007\n"
     ]
    }
   ],
   "source": [
    "# Convert the list to a comma-separated string\n",
    "sample_idx_str = \",\".join(map(str, sample_idx))\n",
    "print((sample_idx_str)[0:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "SELECT ed.sample_index, ed.exon_index, ed.read_count\n",
      "FROM human_disease_b38_gc33.exon_data ed\n",
      "WHERE ed.sample_index IN (85772,85774,85775,85802,85803,85804,85805,85814,85819,85820,85823,85825,85828,85836,85837,85883,85884,85886,85907,85908,85909,85925,85941,85942,85943,85953,85954,85956,85966,85987,85988,85998,86002,86033,86039,86044,86045,86046,86047,220937,220940,220942,220944,220946,220949,220950,220952,220954,220958,220961,220962,220963,220964,220966,220967,220968,220970,220971,220972,220973,220976,220977,220978,220979,220981,220982,220983,220984,220985,220986,220988,220989,220993,220996,220998,221000,221003,221004,221005,221007)\n",
      "\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample_index</th>\n",
       "      <th>exon_index</th>\n",
       "      <th>read_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>85772</td>\n",
       "      <td>12083</td>\n",
       "      <td>4.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>85774</td>\n",
       "      <td>12083</td>\n",
       "      <td>0.040408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>85775</td>\n",
       "      <td>12083</td>\n",
       "      <td>1.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>85802</td>\n",
       "      <td>12083</td>\n",
       "      <td>1.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>85803</td>\n",
       "      <td>12083</td>\n",
       "      <td>1.06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241915</th>\n",
       "      <td>86039</td>\n",
       "      <td>323018</td>\n",
       "      <td>205.4478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241916</th>\n",
       "      <td>86044</td>\n",
       "      <td>323018</td>\n",
       "      <td>170.4055</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241917</th>\n",
       "      <td>86045</td>\n",
       "      <td>323018</td>\n",
       "      <td>114.5631</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241918</th>\n",
       "      <td>86046</td>\n",
       "      <td>323018</td>\n",
       "      <td>52.05908</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32241919</th>\n",
       "      <td>86047</td>\n",
       "      <td>323018</td>\n",
       "      <td>308.0425</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>32241920 rows  3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          sample_index  exon_index  read_count\n",
       "0                85772       12083         4.0\n",
       "1                85774       12083    0.040408\n",
       "2                85775       12083         1.1\n",
       "3                85802       12083        1.06\n",
       "4                85803       12083        1.06\n",
       "...                ...         ...         ...\n",
       "32241915         86039      323018    205.4478\n",
       "32241916         86044      323018    170.4055\n",
       "32241917         86045      323018    114.5631\n",
       "32241918         86046      323018    52.05908\n",
       "32241919         86047      323018    308.0425\n",
       "\n",
       "[32241920 rows x 3 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "sql = f\"\"\"\n",
    "SELECT ed.sample_index, ed.exon_index, ed.read_count\n",
    "FROM human_disease_b38_gc33.exon_data ed\n",
    "WHERE ed.sample_index IN ({sample_idx_str})\n",
    "LIMIT 5\n",
    "\"\"\"\n",
    "print(sql)\n",
    "exon_data = datastore.query(sql)\n",
    "display(exon_data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "False\n",
      "False\n",
      "80\n",
      "80\n"
     ]
    }
   ],
   "source": [
    "# check if there are any duplicate exon indexes per sample index\n",
    "print(exon_data[\"exon_index\"].is_unique)\n",
    "# check if there are any duplicate sample indexes\n",
    "print(exon_data[\"sample_index\"].is_unique)\n",
    "\n",
    "# check if exon indexes are unique when grouped by sample index\n",
    "print(len(exon_data.groupby(\"sample_index\")[\"exon_index\"].nunique()))\n",
    "print(len(exon_data.groupby(\"sample_index\")[\"exon_index\"]))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<IntegerArray>\n",
      "[     0,      1,      2,      3,      4,      5,      6,      7,      8,\n",
      "      9,\n",
      " ...\n",
      " 396435, 396436, 396437, 396438, 396439, 396440, 396441, 396442, 396443,\n",
      " 396444]\n",
      "Length: 396445, dtype: Int32\n"
     ]
    }
   ],
   "source": [
    "exon_index_data = exon_data.sort_values(by=\"exon_index\")\n",
    "print(exon_index_data[\"exon_index\"].unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "sample_index  85772     85774     85775     85802     85803     85804   \\\n",
      "exon_index                                                               \n",
      "0                1.0       1.0      0.74       1.0       1.3      0.92   \n",
      "1              20.68   21.3498  18.64222      16.9      15.8  17.21556   \n",
      "2              16.08  33.94348  16.77778     18.58     20.82  32.75003   \n",
      "3              23.68  24.15652     23.54  24.15167  19.41837  24.72462   \n",
      "4              27.64  25.75652     26.12  27.65167  28.69837  28.02462   \n",
      "\n",
      "sample_index    85805     85814     85819     85820   ...    220988    220989  \\\n",
      "exon_index                                            ...                       \n",
      "0                  0.0       0.0      0.26       0.0  ...  7.037838       0.0   \n",
      "1             23.01188     22.94  12.65347  11.82829  ...  48.95189  109.8223   \n",
      "2             29.51633     27.96     17.28  19.53171  ...  103.3423   131.424   \n",
      "3             23.86512  21.32296     13.88      34.4  ...  68.04674  89.32318   \n",
      "4             26.30512  23.70296     13.88     36.46  ...  79.78007  112.9021   \n",
      "\n",
      "sample_index    220993    220996    220998    221000    221003    221004  \\\n",
      "exon_index                                                                 \n",
      "0                  0.0  2.626667  3.626667       0.0  0.493333  1.568658   \n",
      "1             0.826667     33.24  76.34345   12.1027  77.90414  65.84234   \n",
      "2             3.226667   66.1845  103.8298      24.8  100.4694  120.7919   \n",
      "3             1.573333  49.36272  56.40724  12.54667  67.61676   65.6814   \n",
      "4             1.573333  68.80939  76.78183  22.78667  84.00343   99.7978   \n",
      "\n",
      "sample_index    221005    221007  \n",
      "exon_index                        \n",
      "0             6.951892  3.986667  \n",
      "1             56.22725  92.53402  \n",
      "2             85.80645  142.6787  \n",
      "3             55.73171  101.5641  \n",
      "4             66.89937  137.5775  \n",
      "\n",
      "[5 rows x 80 columns]\n"
     ]
    }
   ],
   "source": [
    "# Pivot table to make count matrix\n",
    "# Pivot the long-format data to create a matrix (samples x exons)\n",
    "count_matrix = exon_data.pivot_table(index='exon_index',\n",
    "                                           columns='sample_index',\n",
    "                                           values='read_count', \n",
    "                                           fill_value=0)\n",
    "print(count_matrix.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "metadata": {},
   "outputs": [],
   "source": [
    "count_matrix.index.name = None\n",
    "count_matrix.columns.name = None\n",
    "\n",
    "print(count_matrix)\n",
    "\n",
    "# Save count matrix df to CSV file\n",
    "count_matrix.to_csv(\"count_matrix_nacc.csv\")\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Extract exon_annotations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "ename": "Exception",
     "evalue": "Unexpected OPI access error: Could not create session: The proxy's IP address(212.237.134.41) is not in the accepted IP range",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mException\u001b[0m                                 Traceback (most recent call last)",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\WebSocketClient.py:158\u001b[0m, in \u001b[0;36mWebSocketClient.configure_auth\u001b[1;34m(self, header, renew_if_device_code)\u001b[0m\n\u001b[0;32m    157\u001b[0m     \u001b[38;5;28;01melif\u001b[39;00m access_response\u001b[38;5;241m.\u001b[39mstatus_code \u001b[38;5;241m==\u001b[39m \u001b[38;5;241m500\u001b[39m:\n\u001b[1;32m--> 158\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(access_response\u001b[38;5;241m.\u001b[39mtext)\n\u001b[0;32m    159\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n",
      "\u001b[1;31mException\u001b[0m: Could not create session: The proxy's IP address(212.237.134.41) is not in the accepted IP range",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[1;31mException\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[24], line 23\u001b[0m\n\u001b[0;32m      1\u001b[0m sql \u001b[38;5;241m=\u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;124mSELECT \u001b[39m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;124m    ea.exon_index,\u001b[39m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     20\u001b[0m \u001b[38;5;124m)\u001b[39m\n\u001b[0;32m     21\u001b[0m \u001b[38;5;124m\"\"\"\u001b[39m\n\u001b[1;32m---> 23\u001b[0m exon_annotation \u001b[38;5;241m=\u001b[39m \u001b[43mdatastore\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mquery\u001b[49m\u001b[43m(\u001b[49m\u001b[43msql\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     24\u001b[0m display(exon_annotation)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\SqlSubmit.py:497\u001b[0m, in \u001b[0;36mSqlSubmit.query\u001b[1;34m(self, sql, databases, default_database, dry_run, show_progress, timeout)\u001b[0m\n\u001b[0;32m    483\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21mquery\u001b[39m(\u001b[38;5;28mself\u001b[39m, sql: \u001b[38;5;28mstr\u001b[39m, \u001b[38;5;241m*\u001b[39m, databases: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m, default_database: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m, dry_run: \u001b[38;5;28mbool\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m, show_progress\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m, timeout: Optional[\u001b[38;5;28mfloat\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m pd\u001b[38;5;241m.\u001b[39mDataFrame:\n\u001b[0;32m    484\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Perform a query and return the result as a DataFrame\u001b[39;00m\n\u001b[0;32m    485\u001b[0m \n\u001b[0;32m    486\u001b[0m \u001b[38;5;124;03m    If databases for some tables in the SQL query are not specified, a unified query will be created to access similar tables across databases. The unified query will be performed across all databases with the necessary tables and the results will be joined together. See the 'opi-python-intro.ipynb' notebook for examples of how to use unified queries.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    495\u001b[0m \u001b[38;5;124;03m    :return: A pandas DataFrame\u001b[39;00m\n\u001b[0;32m    496\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> 497\u001b[0m     query_result \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msubmit_sql\u001b[49m\u001b[43m(\u001b[49m\u001b[43msql\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdatabases\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdatabases\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdefault_database\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdefault_database\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdry_run\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdry_run\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    498\u001b[0m     df \u001b[38;5;241m=\u001b[39m query_result\u001b[38;5;241m.\u001b[39mas_data_frame(show_progress, timeout)\n\u001b[0;32m    499\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m dry_run:\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\SqlSubmit.py:408\u001b[0m, in \u001b[0;36mSqlSubmit.submit_sql\u001b[1;34m(self, sql, compress_download, output_format, default_database, dry_run, databases)\u001b[0m\n\u001b[0;32m    394\u001b[0m \u001b[38;5;28;01mdef\u001b[39;00m \u001b[38;5;21msubmit_sql\u001b[39m(\u001b[38;5;28mself\u001b[39m, sql: \u001b[38;5;28mstr\u001b[39m, \u001b[38;5;241m*\u001b[39m, compress_download: \u001b[38;5;28mbool\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m, output_format: OutputFormat \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m, default_database: \u001b[38;5;28mstr\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124m\"\u001b[39m, dry_run: \u001b[38;5;28mbool\u001b[39m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mFalse\u001b[39;00m, databases: List[\u001b[38;5;28mstr\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m) \u001b[38;5;241m-\u001b[39m\u001b[38;5;241m>\u001b[39m SQLResult:\n\u001b[0;32m    395\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"Submits a SQL query\u001b[39;00m\n\u001b[0;32m    396\u001b[0m \n\u001b[0;32m    397\u001b[0m \u001b[38;5;124;03m    If databases for some tables in the SQL query are not specified, a unified query will be created to access similar tables across databases. The unified query will be performed across all databases with the necessary tables and the results will be joined together. See the 'opi-python-intro.ipynb' notebook for examples of how to use unified queries.\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m    406\u001b[0m \u001b[38;5;124;03m    :returns SQLResult\u001b[39;00m\n\u001b[0;32m    407\u001b[0m \u001b[38;5;124;03m    \"\"\"\u001b[39;00m\n\u001b[1;32m--> 408\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_submit_sql\u001b[49m\u001b[43m(\u001b[49m\u001b[43msql\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mcompress_download\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43moutput_format\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mlands\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdatabases\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdefault_database\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdefault_database\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mdry_run\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mdry_run\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\SqlSubmit.py:477\u001b[0m, in \u001b[0;36mSqlSubmit._submit_sql\u001b[1;34m(self, sql, compress_download, output_format, lands, default_database, dry_run, unified_sql)\u001b[0m\n\u001b[0;32m    470\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m dry_run:\n\u001b[0;32m    471\u001b[0m     payload[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mDryRun\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(\u001b[38;5;28;01mTrue\u001b[39;00m)\n\u001b[0;32m    473\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m SQLResult(\n\u001b[0;32m    474\u001b[0m     sql\u001b[38;5;241m=\u001b[39msql,\n\u001b[0;32m    475\u001b[0m     compress_download\u001b[38;5;241m=\u001b[39mcompress_download,\n\u001b[0;32m    476\u001b[0m     output_format\u001b[38;5;241m=\u001b[39moutput_format,\n\u001b[1;32m--> 477\u001b[0m     future\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_client\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43msubmit\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpayload\u001b[49m\u001b[43m)\u001b[49m,\n\u001b[0;32m    478\u001b[0m     proxy_settings\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dib\u001b[38;5;241m.\u001b[39mget_proxy_settings(),\n\u001b[0;32m    479\u001b[0m     verify_ssl\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dib\u001b[38;5;241m.\u001b[39mget_verify_ssl(),\n\u001b[0;32m    480\u001b[0m     max_search_rows\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_dib\u001b[38;5;241m.\u001b[39mget_max_search_rows()\n\u001b[0;32m    481\u001b[0m )\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\WebSocketClient.py:493\u001b[0m, in \u001b[0;36mWebSocketClient.submit\u001b[1;34m(self, payload)\u001b[0m\n\u001b[0;32m    491\u001b[0m future \u001b[38;5;241m=\u001b[39m ResponseFuture()\n\u001b[0;32m    492\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_requests) \u001b[38;5;241m<\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mmaximum_number_of_concurrent_queries:\n\u001b[1;32m--> 493\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_do_submit\u001b[49m\u001b[43m(\u001b[49m\u001b[43mpayload\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mfuture\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    494\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m future\n\u001b[0;32m    495\u001b[0m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\WebSocketClient.py:600\u001b[0m, in \u001b[0;36mWebSocketClient._do_submit\u001b[1;34m(self, payload, future)\u001b[0m\n\u001b[0;32m    597\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m WebSocketClientException(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mClient already closed\u001b[39m\u001b[38;5;124m\"\u001b[39m)\n\u001b[0;32m    598\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_wsapp \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[0;32m    599\u001b[0m     \u001b[38;5;66;03m# Connection have been closed. Reconnect\u001b[39;00m\n\u001b[1;32m--> 600\u001b[0m     \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_connect\u001b[49m\u001b[43m(\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    601\u001b[0m request_id \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mstr\u001b[39m(random\u001b[38;5;241m.\u001b[39mrandint(\u001b[38;5;241m1\u001b[39m, \u001b[38;5;241m1_000_000_000_000\u001b[39m))\n\u001b[0;32m    602\u001b[0m message \u001b[38;5;241m=\u001b[39m {\n\u001b[0;32m    603\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mAction\u001b[39m\u001b[38;5;124m\"\u001b[39m: \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msubmit\u001b[39m\u001b[38;5;124m\"\u001b[39m,\n\u001b[0;32m    604\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mID\u001b[39m\u001b[38;5;124m\"\u001b[39m: request_id,\n\u001b[0;32m    605\u001b[0m     \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mPayload\u001b[39m\u001b[38;5;124m\"\u001b[39m: payload\n\u001b[0;32m    606\u001b[0m }\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\WebSocketClient.py:205\u001b[0m, in \u001b[0;36mWebSocketClient._connect\u001b[1;34m(self)\u001b[0m\n\u001b[0;32m    203\u001b[0m internal_error_retries \u001b[38;5;241m=\u001b[39m \u001b[38;5;241m0\u001b[39m\n\u001b[0;32m    204\u001b[0m \u001b[38;5;28;01mwhile\u001b[39;00m \u001b[38;5;28;01mTrue\u001b[39;00m:\n\u001b[1;32m--> 205\u001b[0m     header \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mconfigure_auth\u001b[49m\u001b[43m(\u001b[49m\u001b[43mheader\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mrenew_if_device_code\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    206\u001b[0m     logger\u001b[38;5;241m.\u001b[39mdebug(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mWebsocket headers: \u001b[39m\u001b[38;5;132;01m%s\u001b[39;00m\u001b[38;5;124m\"\u001b[39m, header)\n\u001b[0;32m    207\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_wsapp \u001b[38;5;241m=\u001b[39m WebSocketApp(\n\u001b[0;32m    208\u001b[0m         \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_url,\n\u001b[0;32m    209\u001b[0m         header\u001b[38;5;241m=\u001b[39mheader,\n\u001b[0;32m    210\u001b[0m         on_message\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_on_message, on_error\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_on_error, on_close\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_on_close, on_open\u001b[38;5;241m=\u001b[39m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_on_open)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\opi\\dib\\WebSocketClient.py:160\u001b[0m, in \u001b[0;36mWebSocketClient.configure_auth\u001b[1;34m(self, header, renew_if_device_code)\u001b[0m\n\u001b[0;32m    158\u001b[0m             \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(access_response\u001b[38;5;241m.\u001b[39mtext)\n\u001b[0;32m    159\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[1;32m--> 160\u001b[0m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mUnexpected \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mproduct\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m access error: \u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(e)) \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01me\u001b[39;00m\n\u001b[0;32m    161\u001b[0m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mException\u001b[39;00m(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mCould not get session information: \u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;241m+\u001b[39m \u001b[38;5;28mstr\u001b[39m(access_response\u001b[38;5;241m.\u001b[39mstatus_code))\n\u001b[0;32m    163\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_session_cookies_expired \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_session_cookies:\n",
      "\u001b[1;31mException\u001b[0m: Unexpected OPI access error: Could not create session: The proxy's IP address(212.237.134.41) is not in the accepted IP range"
     ]
    }
   ],
   "source": [
    "sql =\"\"\"\n",
    "SELECT \n",
    "    ea.exon_index,\n",
    "    ea.exon_id,\n",
    "    ea.chromosome,\n",
    "    ea.start,\n",
    "    ea.\"end\" AS exon_end,\n",
    "    ea.gene_id,\n",
    "    ea.gene_name,\n",
    "    ea.transcripts\n",
    "FROM human_disease_b38_gc33.exon_annotation ea\n",
    "WHERE ea.exon_index IN (\n",
    "    SELECT DISTINCT ed.exon_index\n",
    "    FROM human_disease_b38_gc33.exon_data ed\n",
    "    JOIN human_disease_b38_gc33.samples s \n",
    "      ON ed.sample_index = s.sample_index\n",
    "    WHERE s.disease_state IN ('schizophrenia', 'normal control', 'disease control', 'schizoaffective disorder')\n",
    "    AND s.tissue = 'nucleus accumbens'\n",
    "    AND s.ethnicity = 'White'\n",
    "    LIMIT 5\n",
    ")\n",
    "\"\"\"\n",
    "\n",
    "exon_annotation = datastore.query(sql)\n",
    "display(exon_annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'exon_annotation' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[23], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# Sanity check if exon indexes in exon_annotations match the merged count matrix\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m \u001b[43mexon_annotation\u001b[49m[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mexon_index\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mnunique()\n",
      "\u001b[1;31mNameError\u001b[0m: name 'exon_annotation' is not defined"
     ]
    }
   ],
   "source": [
    "# Sanity check if exon indexes in exon_annotations match the merged count matrix\n",
    "exon_annotation[\"exon_index\"].nunique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon_index</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>exon_end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcripts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8340</th>\n",
       "      <td>0</td>\n",
       "      <td>1:100000637-100000739</td>\n",
       "      <td>1</td>\n",
       "      <td>100000637</td>\n",
       "      <td>100000739</td>\n",
       "      <td>ENSG00000202259.1</td>\n",
       "      <td>RNU6-1318P</td>\n",
       "      <td>ENST00000365389.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270634</th>\n",
       "      <td>1</td>\n",
       "      <td>1:100007034-100007156</td>\n",
       "      <td>1</td>\n",
       "      <td>100007034</td>\n",
       "      <td>100007156</td>\n",
       "      <td>ENSG00000117620.15</td>\n",
       "      <td>SLC35A3</td>\n",
       "      <td>ENST00000638371.1;ENST00000427993.7;ENST000005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>270635</th>\n",
       "      <td>1</td>\n",
       "      <td>1:100007034-100007156</td>\n",
       "      <td>1</td>\n",
       "      <td>100007034</td>\n",
       "      <td>100007156</td>\n",
       "      <td>ENSG00000283761.1</td>\n",
       "      <td>AC118553.2</td>\n",
       "      <td>ENST00000638792.1;ENST00000639037.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137220</th>\n",
       "      <td>2</td>\n",
       "      <td>1:100011365-100011533</td>\n",
       "      <td>1</td>\n",
       "      <td>100011365</td>\n",
       "      <td>100011533</td>\n",
       "      <td>ENSG00000283761.1</td>\n",
       "      <td>AC118553.2</td>\n",
       "      <td>ENST00000638792.1;ENST00000639037.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137219</th>\n",
       "      <td>2</td>\n",
       "      <td>1:100011365-100011533</td>\n",
       "      <td>1</td>\n",
       "      <td>100011365</td>\n",
       "      <td>100011533</td>\n",
       "      <td>ENSG00000117620.15</td>\n",
       "      <td>SLC35A3</td>\n",
       "      <td>ENST00000638371.1;ENST00000427993.7;ENST000005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79888</th>\n",
       "      <td>396440</td>\n",
       "      <td>Y:9958419-9958529</td>\n",
       "      <td>Y</td>\n",
       "      <td>9958419</td>\n",
       "      <td>9958529</td>\n",
       "      <td>ENSG00000234950.1</td>\n",
       "      <td>RBMY2OP</td>\n",
       "      <td>ENST00000447105.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>334210</th>\n",
       "      <td>396441</td>\n",
       "      <td>Y:9958610-9958698</td>\n",
       "      <td>Y</td>\n",
       "      <td>9958610</td>\n",
       "      <td>9958698</td>\n",
       "      <td>ENSG00000234950.1</td>\n",
       "      <td>RBMY2OP</td>\n",
       "      <td>ENST00000447105.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>352553</th>\n",
       "      <td>396442</td>\n",
       "      <td>Y:9959112-9959423</td>\n",
       "      <td>Y</td>\n",
       "      <td>9959112</td>\n",
       "      <td>9959423</td>\n",
       "      <td>ENSG00000234950.1</td>\n",
       "      <td>RBMY2OP</td>\n",
       "      <td>ENST00000447105.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>389575</th>\n",
       "      <td>396443</td>\n",
       "      <td>Y:9989190-9989239</td>\n",
       "      <td>Y</td>\n",
       "      <td>9989190</td>\n",
       "      <td>9989239</td>\n",
       "      <td>ENSG00000236718.4</td>\n",
       "      <td>RBMY2QP</td>\n",
       "      <td>ENST00000651645.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70643</th>\n",
       "      <td>396444</td>\n",
       "      <td>Y:9995597-9995715</td>\n",
       "      <td>Y</td>\n",
       "      <td>9995597</td>\n",
       "      <td>9995715</td>\n",
       "      <td>ENSG00000236718.4</td>\n",
       "      <td>RBMY2QP</td>\n",
       "      <td>ENST00000651645.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>403024 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        exon_index                exon_id chromosome      start   exon_end  \\\n",
       "8340             0  1:100000637-100000739          1  100000637  100000739   \n",
       "270634           1  1:100007034-100007156          1  100007034  100007156   \n",
       "270635           1  1:100007034-100007156          1  100007034  100007156   \n",
       "137220           2  1:100011365-100011533          1  100011365  100011533   \n",
       "137219           2  1:100011365-100011533          1  100011365  100011533   \n",
       "...            ...                    ...        ...        ...        ...   \n",
       "79888       396440      Y:9958419-9958529          Y    9958419    9958529   \n",
       "334210      396441      Y:9958610-9958698          Y    9958610    9958698   \n",
       "352553      396442      Y:9959112-9959423          Y    9959112    9959423   \n",
       "389575      396443      Y:9989190-9989239          Y    9989190    9989239   \n",
       "70643       396444      Y:9995597-9995715          Y    9995597    9995715   \n",
       "\n",
       "                   gene_id   gene_name  \\\n",
       "8340     ENSG00000202259.1  RNU6-1318P   \n",
       "270634  ENSG00000117620.15     SLC35A3   \n",
       "270635   ENSG00000283761.1  AC118553.2   \n",
       "137220   ENSG00000283761.1  AC118553.2   \n",
       "137219  ENSG00000117620.15     SLC35A3   \n",
       "...                    ...         ...   \n",
       "79888    ENSG00000234950.1     RBMY2OP   \n",
       "334210   ENSG00000234950.1     RBMY2OP   \n",
       "352553   ENSG00000234950.1     RBMY2OP   \n",
       "389575   ENSG00000236718.4     RBMY2QP   \n",
       "70643    ENSG00000236718.4     RBMY2QP   \n",
       "\n",
       "                                              transcripts  \n",
       "8340                                    ENST00000365389.1  \n",
       "270634  ENST00000638371.1;ENST00000427993.7;ENST000005...  \n",
       "270635                ENST00000638792.1;ENST00000639037.1  \n",
       "137220                ENST00000638792.1;ENST00000639037.1  \n",
       "137219  ENST00000638371.1;ENST00000427993.7;ENST000005...  \n",
       "...                                                   ...  \n",
       "79888                                   ENST00000447105.1  \n",
       "334210                                  ENST00000447105.1  \n",
       "352553                                  ENST00000447105.1  \n",
       "389575                                  ENST00000651645.1  \n",
       "70643                                   ENST00000651645.1  \n",
       "\n",
       "[403024 rows x 8 columns]"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# sort indexes by exon_index\n",
    "exon_annotation = exon_annotation.sort_values(by=\"exon_index\")\n",
    "exon_annotation.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save to csv\n",
    "exon_annotation.to_csv(\"exon_annotation_nacc.csv\", index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sanity checking the resulting data - get back to this"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Count matrix shape: (396445, 80)\n",
      "Row index (exon IDs) type: Int32\n",
      "Column names (sample IDs) type: Int32\n",
      "   85772     85774     85775     85802     85803     85804     85805   \\\n",
      "0     1.0       1.0      0.74       1.0       1.3      0.92       0.0   \n",
      "1   20.68   21.3498  18.64222      16.9      15.8  17.21556  23.01188   \n",
      "2   16.08  33.94348  16.77778     18.58     20.82  32.75003  29.51633   \n",
      "3   23.68  24.15652     23.54  24.15167  19.41837  24.72462  23.86512   \n",
      "4   27.64  25.75652     26.12  27.65167  28.69837  28.02462  26.30512   \n",
      "\n",
      "     85814     85819     85820   ...    220988    220989    220993    220996  \\\n",
      "0       0.0      0.26       0.0  ...  7.037838       0.0       0.0  2.626667   \n",
      "1     22.94  12.65347  11.82829  ...  48.95189  109.8223  0.826667     33.24   \n",
      "2     27.96     17.28  19.53171  ...  103.3423   131.424  3.226667   66.1845   \n",
      "3  21.32296     13.88      34.4  ...  68.04674  89.32318  1.573333  49.36272   \n",
      "4  23.70296     13.88     36.46  ...  79.78007  112.9021  1.573333  68.80939   \n",
      "\n",
      "     220998    221000    221003    221004    221005    221007  \n",
      "0  3.626667       0.0  0.493333  1.568658  6.951892  3.986667  \n",
      "1  76.34345   12.1027  77.90414  65.84234  56.22725  92.53402  \n",
      "2  103.8298      24.8  100.4694  120.7919  85.80645  142.6787  \n",
      "3  56.40724  12.54667  67.61676   65.6814  55.73171  101.5641  \n",
      "4  76.78183  22.78667  84.00343   99.7978  66.89937  137.5775  \n",
      "\n",
      "[5 rows x 80 columns]\n"
     ]
    }
   ],
   "source": [
    "### Sanity checking data\n",
    "\n",
    "# Checking data types and dimensions\n",
    "print(\"Count matrix shape:\", count_matrix.shape)\n",
    "print(\"Row index (exon IDs) type:\", count_matrix.index.dtype)\n",
    "print(\"Column names (sample IDs) type:\", count_matrix.columns.dtype)\n",
    "\n",
    "# Inspect the first few rows and columns\n",
    "print(count_matrix.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of sample IDs in count matrix: 80\n",
      "Number of sample IDs in metadata: 80\n",
      "All sample IDs in count matrix are present in metadata.\n"
     ]
    }
   ],
   "source": [
    "# Get sample indexes from the count matrix columns and metadata\n",
    "sample_idx_from_counts = set(count_matrix.columns.tolist())\n",
    "sample_idx_from_metadata = set(metadata_nacc[\"sample_index\"].tolist())\n",
    "\n",
    "print(\"Number of sample_indexes in count matrix:\", len(sample_idx_from_counts))\n",
    "print(\"Number of sample_indexes in metadata:\", len(sample_idx_from_metadata))\n",
    "\n",
    "# Check if all sample IDs in the count matrix are in the metadata:\n",
    "missing_in_metadata = sample_idx_from_counts.difference(sample_idx_from_metadata)\n",
    "if missing_in_metadata:\n",
    "    print(\"Sample indices missing in metadata:\", missing_in_metadata)\n",
    "else:\n",
    "    print(\"All sample indexes in count matrix are present in metadata.\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of exon_indexes in count matrix: 396445\n",
      "Number of exon_indexes in annotation: 396445\n",
      "All exon indexes in count matrix are present in annotation.\n"
     ]
    }
   ],
   "source": [
    "# Check if exon indexes in count matrix are in exon_annotation\n",
    "exon_idx_from_counts = set(count_matrix.index.tolist())\n",
    "exon_idx_from_annotation = set(exon_annotation[\"exon_index\"].tolist())\n",
    "print(\"Number of exon_indexes in count matrix:\", len(exon_idx_from_counts))\n",
    "print(\"Number of exon_indexes in annotation:\", len(exon_idx_from_annotation))\n",
    "# Check if all exon indexes in the count matrix are in the annotation:\n",
    "missing_in_annotation = exon_idx_from_counts.difference(exon_idx_from_annotation)\n",
    "if missing_in_annotation:\n",
    "    print(\"Exon indexes missing in annotation:\", missing_in_annotation)\n",
    "else:\n",
    "    print(\"All exon indexes in count matrix are present in annotation.\")\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "disease_state\n",
       "normal control    48\n",
       "schizophrenia     32\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 180,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check number of samples in each disease condition\n",
    "metadata_nacc[\"disease_state\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'metadata_nacc' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[7], line 2\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# Group by title and check disease state c\u001b[39;00m\n\u001b[1;32m----> 2\u001b[0m \u001b[43mmetadata_nacc\u001b[49m\u001b[38;5;241m.\u001b[39mgroupby([\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mtitle\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mdisease_state\u001b[39m\u001b[38;5;124m\"\u001b[39m])[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124msample_index\u001b[39m\u001b[38;5;124m\"\u001b[39m]\u001b[38;5;241m.\u001b[39mcount()\n",
      "\u001b[1;31mNameError\u001b[0m: name 'metadata_nacc' is not defined"
     ]
    }
   ],
   "source": [
    "# Group by title and check disease state c\n",
    "metadata_nacc.groupby([\"title\", \"disease_state\"])[\"sample_index\"].count()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "tissue\n",
       "nucleus accumbens    80\n",
       "Name: count, dtype: Int64"
      ]
     },
     "execution_count": 181,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Check number of samples in each tissue\n",
    "metadata_nacc[\"tissue\"].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "exon_index\n",
       "297498    22\n",
       "297497    22\n",
       "297336    15\n",
       "297335    15\n",
       "297504    14\n",
       "          ..\n",
       "396363     1\n",
       "396374     1\n",
       "396390     1\n",
       "396427     1\n",
       "125303     1\n",
       "Name: count, Length: 396445, dtype: Int64"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# There are duplicate exon indexes (rows) in exon_annotations (same exon is used by multiple genes)\n",
    "dup_counts = exon_annotation['exon_index'].value_counts()\n",
    "dup_counts[1:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>exon_index</th>\n",
       "      <th>exon_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>exon_end</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>gene_name</th>\n",
       "      <th>transcripts</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td>131129</td>\n",
       "      <td>15:89889049-89889140</td>\n",
       "      <td>15</td>\n",
       "      <td>89889049</td>\n",
       "      <td>89889140</td>\n",
       "      <td>ENSG00000157823.17</td>\n",
       "      <td>AP3S2</td>\n",
       "      <td>ENST00000336418.9;ENST00000558011.5;ENST000005...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td>131129</td>\n",
       "      <td>15:89889049-89889140</td>\n",
       "      <td>15</td>\n",
       "      <td>89889049</td>\n",
       "      <td>89889140</td>\n",
       "      <td>ENSG00000250021.7</td>\n",
       "      <td>ARPIN-AP3S2</td>\n",
       "      <td>ENST00000398333.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>292</th>\n",
       "      <td>133746</td>\n",
       "      <td>16:15118926-15119054</td>\n",
       "      <td>16</td>\n",
       "      <td>15118926</td>\n",
       "      <td>15119054</td>\n",
       "      <td>ENSG00000188599.17</td>\n",
       "      <td>NPIPP1</td>\n",
       "      <td>ENST00000534799.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>293</th>\n",
       "      <td>133746</td>\n",
       "      <td>16:15118926-15119054</td>\n",
       "      <td>16</td>\n",
       "      <td>15118926</td>\n",
       "      <td>15119054</td>\n",
       "      <td>ENSG00000270580.5</td>\n",
       "      <td>AC139256.3</td>\n",
       "      <td>ENST00000605794.5;ENST00000340301.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>304</th>\n",
       "      <td>134224</td>\n",
       "      <td>16:1662761-1662835</td>\n",
       "      <td>16</td>\n",
       "      <td>1662761</td>\n",
       "      <td>1662835</td>\n",
       "      <td>ENSG00000007545.16</td>\n",
       "      <td>CRAMP1</td>\n",
       "      <td>ENST00000397412.8;ENST00000293925.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402706</th>\n",
       "      <td>386677</td>\n",
       "      <td>X:15664337-15664395</td>\n",
       "      <td>X</td>\n",
       "      <td>15664337</td>\n",
       "      <td>15664395</td>\n",
       "      <td>ENSG00000147003.7</td>\n",
       "      <td>CLTRN</td>\n",
       "      <td>ENST00000380342.4;ENST00000650271.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402766</th>\n",
       "      <td>388474</td>\n",
       "      <td>X:38666121-38666309</td>\n",
       "      <td>X</td>\n",
       "      <td>38666121</td>\n",
       "      <td>38666309</td>\n",
       "      <td>ENSG00000250349.3</td>\n",
       "      <td>AF241726.2</td>\n",
       "      <td>ENST00000465127.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402767</th>\n",
       "      <td>388474</td>\n",
       "      <td>X:38666121-38666309</td>\n",
       "      <td>X</td>\n",
       "      <td>38666121</td>\n",
       "      <td>38666309</td>\n",
       "      <td>ENSG00000156298.12</td>\n",
       "      <td>TSPAN7</td>\n",
       "      <td>ENST00000378482.6;ENST00000286824.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402849</th>\n",
       "      <td>391206</td>\n",
       "      <td>X:55451495-55451582</td>\n",
       "      <td>X</td>\n",
       "      <td>55451495</td>\n",
       "      <td>55451582</td>\n",
       "      <td>MI0016906</td>\n",
       "      <td>hsa-mir-4536-1</td>\n",
       "      <td>t_MI0016906</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>402850</th>\n",
       "      <td>391206</td>\n",
       "      <td>X:55451495-55451582</td>\n",
       "      <td>X</td>\n",
       "      <td>55451495</td>\n",
       "      <td>55451582</td>\n",
       "      <td>MI0019149</td>\n",
       "      <td>hsa-mir-4536-2</td>\n",
       "      <td>t_MI0019149</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>12867 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        exon_index               exon_id chromosome     start  exon_end  \\\n",
       "203         131129  15:89889049-89889140         15  89889049  89889140   \n",
       "204         131129  15:89889049-89889140         15  89889049  89889140   \n",
       "292         133746  16:15118926-15119054         16  15118926  15119054   \n",
       "293         133746  16:15118926-15119054         16  15118926  15119054   \n",
       "304         134224    16:1662761-1662835         16   1662761   1662835   \n",
       "...            ...                   ...        ...       ...       ...   \n",
       "402706      386677   X:15664337-15664395          X  15664337  15664395   \n",
       "402766      388474   X:38666121-38666309          X  38666121  38666309   \n",
       "402767      388474   X:38666121-38666309          X  38666121  38666309   \n",
       "402849      391206   X:55451495-55451582          X  55451495  55451582   \n",
       "402850      391206   X:55451495-55451582          X  55451495  55451582   \n",
       "\n",
       "                   gene_id       gene_name  \\\n",
       "203     ENSG00000157823.17           AP3S2   \n",
       "204      ENSG00000250021.7     ARPIN-AP3S2   \n",
       "292     ENSG00000188599.17          NPIPP1   \n",
       "293      ENSG00000270580.5      AC139256.3   \n",
       "304     ENSG00000007545.16          CRAMP1   \n",
       "...                    ...             ...   \n",
       "402706   ENSG00000147003.7           CLTRN   \n",
       "402766   ENSG00000250349.3      AF241726.2   \n",
       "402767  ENSG00000156298.12          TSPAN7   \n",
       "402849           MI0016906  hsa-mir-4536-1   \n",
       "402850           MI0019149  hsa-mir-4536-2   \n",
       "\n",
       "                                              transcripts  \n",
       "203     ENST00000336418.9;ENST00000558011.5;ENST000005...  \n",
       "204                                     ENST00000398333.7  \n",
       "292                                     ENST00000534799.6  \n",
       "293                   ENST00000605794.5;ENST00000340301.5  \n",
       "304                   ENST00000397412.8;ENST00000293925.9  \n",
       "...                                                   ...  \n",
       "402706                ENST00000380342.4;ENST00000650271.1  \n",
       "402766                                  ENST00000465127.1  \n",
       "402767                ENST00000378482.6;ENST00000286824.6  \n",
       "402849                                        t_MI0016906  \n",
       "402850                                        t_MI0019149  \n",
       "\n",
       "[12867 rows x 8 columns]"
      ]
     },
     "execution_count": 183,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# show the duplicated rows\n",
    "duplicated_exons = exon_annotation[exon_annotation['exon_index'].isin(dup_counts[dup_counts > 1].index)]\n",
    "duplicated_exons.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Distribution of exon total counts:\n",
      "count         396445.0\n",
      "mean      23485.485747\n",
      "std      721345.764226\n",
      "min                0.0\n",
      "25%         177.255931\n",
      "50%         1812.43165\n",
      "75%         8325.54946\n",
      "max      220827033.636\n",
      "dtype: Float64\n",
      "Filtered count matrix shape: (374542, 80)\n"
     ]
    }
   ],
   "source": [
    "# Sum counts per exon (sums all read count values across each exon)\n",
    "exon_counts = count_matrix.sum(axis=1)\n",
    "print(\"Distribution of exon total counts:\")\n",
    "print(exon_counts.describe())\n",
    "\n",
    "# Optionally, filter out exons with total counts below a threshold (e.g., 10) - we get back to this later\n",
    "threshold = 10\n",
    "filtered_count_matrix_df = count_matrix.loc[exon_counts >= threshold]\n",
    "print(\"Filtered count matrix shape:\", filtered_count_matrix_df.shape)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Check unique genes\n",
    "unique_genes = exon_annotation['gene_name'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "60615\n",
      "60699\n"
     ]
    }
   ],
   "source": [
    "print(len(unique_genes))\n",
    "\n",
    "print(len(exon_annotation['gene_id'].unique()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "IDs with NO name at all: 0\n",
      "Examples: []\n",
      "Names with NO id at all: 0\n",
      "Examples: []\n"
     ]
    }
   ],
   "source": [
    "# Explore if any gene names or ID don't have a name or ID\n",
    "ids_has_any_name = exon_annotation.groupby(\"gene_id\", dropna=True)[\"gene_name\"].apply(lambda s: s.notna().any())\n",
    "ids_without_name = ids_has_any_name[ids_has_any_name == False].index.tolist()\n",
    "\n",
    "# Names that never have a non-null gene_id anywhere\n",
    "names_has_any_id = exon_annotation.groupby(\"gene_name\", dropna=True)[\"gene_id\"].apply(lambda s: s.notna().any())\n",
    "names_without_id = names_has_any_id[names_has_any_id == False].index.tolist()\n",
    "\n",
    "print(\"\\nIDs with NO name at all:\", len(ids_without_name))\n",
    "print(\"Examples:\", ids_without_name[:10])\n",
    "print(\"Names with NO id at all:\", len(names_without_id))\n",
    "print(\"Examples:\", names_without_id[:10])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Gene names mapping to >1 gene_id: 82\n",
      "gene_name\n",
      "hsa-mir-10401    4\n",
      "AL732314.8       2\n",
      "ASMTL            2\n",
      "ASMTL-AS1        2\n",
      "AL954722.1       2\n",
      "                ..\n",
      "AKAP17A          2\n",
      "hsa-mir-1253     2\n",
      "hsa-mir-423      2\n",
      "hsa-mir-4477a    2\n",
      "hsa-mir-4477b    2\n",
      "Name: gene_id, Length: 82, dtype: int64\n",
      "\n",
      "Gene IDs mapping to >1 gene_name: 0\n",
      "Series([], Name: gene_name, dtype: int64)\n"
     ]
    }
   ],
   "source": [
    "# Check names that map to multiple IDs\n",
    "name_to_ids_n = (exon_annotation.dropna(subset=[\"gene_id\",\"gene_name\"])\n",
    "                   .groupby(\"gene_name\")[\"gene_id\"].nunique()\n",
    "                   .sort_values(ascending=False))\n",
    "names_multi_ids = name_to_ids_n[name_to_ids_n > 1]\n",
    "\n",
    "# IDs that map to multiple Names\n",
    "id_to_names_n = (exon_annotation.dropna(subset=[\"gene_id\",\"gene_name\"])\n",
    "                   .groupby(\"gene_id\")[\"gene_name\"].nunique()\n",
    "                   .sort_values(ascending=False))\n",
    "ids_multi_names = id_to_names_n[id_to_names_n > 1]\n",
    "\n",
    "print(\"\\nGene names mapping to >1 gene_id:\", len(names_multi_ids))\n",
    "names_multi_ids.asDataFrame()\n",
    "print(names_multi_ids)\n",
    "print(\"\\nGene IDs mapping to >1 gene_name:\", len(ids_multi_names))\n",
    "print(ids_multi_names.head(10))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "Rows with ID but no NAME: 0\n",
      "Empty DataFrame\n",
      "Columns: [gene_id, gene_name]\n",
      "Index: []\n",
      "\n",
      "Rows with NAME but no ID: 0\n",
      "Empty DataFrame\n",
      "Columns: [gene_id, gene_name]\n",
      "Index: []\n"
     ]
    }
   ],
   "source": [
    "# Direct orphan rows: where one side is present but the other is null\n",
    "orphan_id_rows = exon_annotation[exon_annotation[\"gene_id\"].notna() & exon_annotation[\"gene_name\"].isna()]\n",
    "orphan_name_rows = exon_annotation[exon_annotation[\"gene_name\"].notna() & exon_annotation[\"gene_id\"].isna()]\n",
    "\n",
    "print(\"\\nRows with ID but no NAME:\", orphan_id_rows.shape[0])\n",
    "print(orphan_id_rows.head(5)[[\"gene_id\",\"gene_name\"]])\n",
    "print(\"\\nRows with NAME but no ID:\", orphan_name_rows.shape[0])\n",
    "print(orphan_name_rows.head(5)[[\"gene_id\",\"gene_name\"]])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "If need to read data in"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "ename": "RuntimeError",
     "evalue": "CPU dispatcher tracer already initlized",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mRuntimeError\u001b[0m                              Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[8], line 1\u001b[0m\n\u001b[1;32m----> 1\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mpandas\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m \u001b[38;5;21;01mpd\u001b[39;00m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;66;03m# read in metadata_nacc\u001b[39;00m\n\u001b[0;32m      3\u001b[0m metadata_nacc \u001b[38;5;241m=\u001b[39m pd\u001b[38;5;241m.\u001b[39mread_csv(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mmetadata_nacc_.csv\u001b[39m\u001b[38;5;124m'\u001b[39m)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\pandas\\__init__.py:27\u001b[0m\n\u001b[0;32m     25\u001b[0m \u001b[38;5;28;01mfor\u001b[39;00m _dependency \u001b[38;5;129;01min\u001b[39;00m _hard_dependencies:\n\u001b[0;32m     26\u001b[0m     \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m---> 27\u001b[0m         \u001b[38;5;28;43m__import__\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43m_dependency\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m     28\u001b[0m     \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m _e:  \u001b[38;5;66;03m# pragma: no cover\u001b[39;00m\n\u001b[0;32m     29\u001b[0m         _missing_dependencies\u001b[38;5;241m.\u001b[39mappend(\u001b[38;5;124mf\u001b[39m\u001b[38;5;124m\"\u001b[39m\u001b[38;5;132;01m{\u001b[39;00m_dependency\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m: \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m_e\u001b[38;5;132;01m}\u001b[39;00m\u001b[38;5;124m\"\u001b[39m)\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\numpy\\__init__.py:128\u001b[0m\n\u001b[0;32m    125\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m _distributor_init\n\u001b[0;32m    127\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m--> 128\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m__config__\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m show \u001b[38;5;28;01mas\u001b[39;00m show_config\n\u001b[0;32m    129\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m e:\n\u001b[0;32m    130\u001b[0m     msg \u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\"\"\u001b[39m\u001b[38;5;124mError importing numpy: you should not try to import numpy from\u001b[39m\n\u001b[0;32m    131\u001b[0m \u001b[38;5;124m    its source directory; please exit the numpy source tree, and relaunch\u001b[39m\n\u001b[0;32m    132\u001b[0m \u001b[38;5;124m    your python interpreter from there.\u001b[39m\u001b[38;5;124m\"\"\"\u001b[39m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\numpy\\__config__.py:4\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# This file is generated by numpy's build process\u001b[39;00m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;66;03m# It contains system_info results at the time of building this package.\u001b[39;00m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01menum\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m Enum\n\u001b[1;32m----> 4\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_core\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_multiarray_umath\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[0;32m      5\u001b[0m     __cpu_features__,\n\u001b[0;32m      6\u001b[0m     __cpu_baseline__,\n\u001b[0;32m      7\u001b[0m     __cpu_dispatch__,\n\u001b[0;32m      8\u001b[0m )\n\u001b[0;32m     10\u001b[0m __all__ \u001b[38;5;241m=\u001b[39m [\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mshow\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m     11\u001b[0m _built_with_meson \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mTrue\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\numpy\\_core\\__init__.py:23\u001b[0m\n\u001b[0;32m     20\u001b[0m         env_added\u001b[38;5;241m.\u001b[39mappend(envkey)\n\u001b[0;32m     22\u001b[0m \u001b[38;5;28;01mtry\u001b[39;00m:\n\u001b[1;32m---> 23\u001b[0m     \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m multiarray\n\u001b[0;32m     24\u001b[0m \u001b[38;5;28;01mexcept\u001b[39;00m \u001b[38;5;167;01mImportError\u001b[39;00m \u001b[38;5;28;01mas\u001b[39;00m exc:\n\u001b[0;32m     25\u001b[0m     \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01msys\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\numpy\\_core\\multiarray.py:10\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m      2\u001b[0m \u001b[38;5;124;03mCreate the numpy._core.multiarray namespace for backward compatibility. \u001b[39;00m\n\u001b[0;32m      3\u001b[0m \u001b[38;5;124;03mIn v1.16 the multiarray and umath c-extension modules were merged into \u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m      6\u001b[0m \n\u001b[0;32m      7\u001b[0m \u001b[38;5;124;03m\"\"\"\u001b[39;00m\n\u001b[0;32m      9\u001b[0m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;21;01mfunctools\u001b[39;00m\n\u001b[1;32m---> 10\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m overrides\n\u001b[0;32m     11\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m _multiarray_umath\n\u001b[0;32m     12\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_multiarray_umath\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m \u001b[38;5;241m*\u001b[39m  \u001b[38;5;66;03m# noqa: F403\u001b[39;00m\n",
      "File \u001b[1;32m~\\AppData\\Local\\Packages\\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\\LocalCache\\local-packages\\Python311\\site-packages\\numpy\\_core\\overrides.py:8\u001b[0m\n\u001b[0;32m      6\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_utils\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m set_module\n\u001b[0;32m      7\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_utils\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_inspect\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m getargspec\n\u001b[1;32m----> 8\u001b[0m \u001b[38;5;28;01mfrom\u001b[39;00m \u001b[38;5;21;01mnumpy\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_core\u001b[39;00m\u001b[38;5;21;01m.\u001b[39;00m\u001b[38;5;21;01m_multiarray_umath\u001b[39;00m \u001b[38;5;28;01mimport\u001b[39;00m (\n\u001b[0;32m      9\u001b[0m     add_docstring,  _get_implementing_args, _ArrayFunctionDispatcher)\n\u001b[0;32m     12\u001b[0m ARRAY_FUNCTIONS \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mset\u001b[39m()\n\u001b[0;32m     14\u001b[0m array_function_like_doc \u001b[38;5;241m=\u001b[39m (\n\u001b[0;32m     15\u001b[0m \u001b[38;5;250m    \u001b[39m\u001b[38;5;124;03m\"\"\"like : array_like, optional\u001b[39;00m\n\u001b[0;32m     16\u001b[0m \u001b[38;5;124;03m        Reference object to allow the creation of arrays which are not\u001b[39;00m\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m     20\u001b[0m \u001b[38;5;124;03m        compatible with that passed in via this argument.\"\"\"\u001b[39;00m\n\u001b[0;32m     21\u001b[0m )\n",
      "\u001b[1;31mRuntimeError\u001b[0m: CPU dispatcher tracer already initlized"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "# read in metadata_nacc\n",
    "metadata_nacc = pd.read_csv('metadata_nacc_.csv')\n",
    "\n",
    "# Read in count matrix\n",
    "count_matrix = pd.read_csv(\"count_matrix_nacc.csv\")\n",
    "print(count_matrix.index) \n",
    "\n",
    "\n",
    "# Import exon annotations\n",
    "exon_annotation = pd.read_csv(\"input_data/combined_study/exon_annotation_nacc.csv\", low_memory = False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.11.9"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
